app/src/main/kotlin/at/bitfire/davdroid/ui/widget/RadioWithSwitch.kt

```diff
@@ -0,0 +1,178 @@
+                        )
+                    ),
+                    modifier = Modifier.fillMaxWidth(),
+                    onClick = { index ->
+                        // Get the tapped position, and check if there's any link
+                        val annotation = sum.getUrlAnnotations(index, index).firstOrNull()
+                        try {
+                            // If there is, open it
+                            annotation?.item?.url?.let(uriHandler::openUri)
+                        } catch (_: ActivityNotFoundException) {
+                            // There isn't any application available to launch the link
+                            Log.e("RadioWithSwitch", "No app available to launch ${annotation?.item?.url}")
+                        }
+                    }
+                )
+            }
+        }
+
+        Switch(
+            checked = isToggled,
+            onCheckedChange = onToggled,
+            enabled = !enabled
+        )
+    }
+}
+
+/**
+ * Provides a radio button with a text, a switch at the end, and an optional summary to be shown
+ * under the main text.
+ *
+ * @param title The "proper" text of the Radio button. Shown in the middle of the row, between the
+ * radio button and the switch.
+ * @param summary If not `null`, shown below the title. Used to give more context or information.
+ * Supports formatting and interactions.
+ * @param isSelected Whether the item is currently selected. Refers to the radio button.
+ * @param isToggled Whether the switch is toggled.
+ * @param modifier Any modifiers to apply to the row.
+ * @param enabled Whether the radio button should be enabled. The enabled state of the switch is
+ * reverse from this. So if it's `true`, the switch will be disabled.
+ * @param onSelected Gets called whenever the user requests this row to be enabled. Either by
+ * selecting the radio button or tapping the text.
+ * @param onToggled Gets called whenever the switch gets updated. Contains the checked status.
+ */
+@Composable
+@ExperimentalTextApi
+fun RadioWithSwitch(
+    title: String,
+    summary: String?,
+    isSelected: Boolean,
+    isToggled: Boolean,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    onSelected: () -> Unit,
+    onToggled: (Boolean) -> Unit
+) {
+    RadioWithSwitch(
+        title,
+        buildAnnotatedString { append(summary) },
+        isSelected,
+        isToggled,
+        modifier,
+        enabled,
+        onSelected,
+        onToggled
+    )
+}
+
+private class PreviewProvider : PreviewParameterProvider<PreviewProvider.PreviewData> {
+    data class PreviewData(
+        val title: String,
+        val summary: String?,
+        val isSelected: Boolean,
+        val isToggled: Boolean
+    )
+
+    override val values: Sequence<PreviewData> = sequenceOf(
+        PreviewData("RadioWithSwitch Preview", "An example summary", true, isToggled = false)
+    )
+}
```

Comment:
Is there an advantage of creating the preview providere here? For me it's easier to understand at the first glance if the `@Preview` just contains the 4 parameters.

Or if we want to show multiple combinations of isSelected, isToggled usw, I understand what we need the PreviewProvider for. But then we should show these combinations.

Reply:
The idea was to provide all the possible combinations, that's why the preview provider exists, but I ended up removing them because the preview was a bit cluttered, so yeah, if we don't want more previews, we can just get rid of this.